src/runtime/virtual/server-handlers.d.ts

```diff
@@ -1,11 +1,17 @@
-import type { CompatibilityEventHandler, RouterMethod } from 'h3'
+import type { CompatibilityEventHandler, LazyEventHandler, RouterMethod } from 'h3'
 
-interface HandlerDefinition {
+type HandlerDefinition = {
   route: string
-  lazy?: boolean
   middleware?: boolean
-  handler: CompatibilityEventHandler | (() => Promise<CompatibilityEventHandler>)
   method?: RouterMethod
 }
+& {
+  lazy?: true
+  handler: LazyEventHandler | (() => Promise<LazyEventHandler>)
+}
+& {
+  lazy?: false
+  handler: CompatibilityEventHandler | (() => Promise<CompatibilityEventHandler>)
+}
```

Comment:
It seems reduction is not working as lazy is optional: 

<img width="657" alt="image" src="https://user-images.githubusercontent.com/5158436/173253383-65af003b-8cac-439c-bc0d-5dc0271c2f0b.png">

Reply:
(updated: added back as a more compact type)